components/customs/multi-select.tsx

```diff
@@ -126,7 +126,10 @@ export function MultiSelect(props: MultiSelectProps) {
                       onSelect={(value) => {
                         setInputValue("");
                         const newValue = selectables.find(
-                          (option) => option.value === value,
+                          // CommandItem appears to strip starting and ending whitespace from the value given as a key
+                          // so the trim() fixes a situation where the value has some starting or trailing whitespace
+                          // and isn't recognized as a selectable option as a result.
+                          (option) => option.value.trim() === value,
```

Comment:
For those who don't understand, the underling value for "Euroimmun" is `euroimmun ` and the underlying component library strips away whitespace when it sends it's `onSelect` event for some reason. All our logic still works with `euroimmun `, we just need to wrap this one particular event handler basically.

We could technically strip away the whitespace in the backend but it's kind of silly that you can't have any extra whitespace in your producer name imo so we should ship this anyways. I'm not sure if it's a super good use of our time to strip the whitespace on the backend but I could definitely make a ticket and we could hit it if there's interest.